src/elf.rs

```diff
@@ -719,8 +727,81 @@ impl<'a> Elf<'a> {
             self.nsections += 1;
         }
     }
+    pub fn add_extended_symtab(&mut self) -> usize {
+        // copypasta'd from add_progbits
+        let name = ".symtab_shndxr".to_string();
+        let (idx, offset) = self.new_string(name);
+        debug!(
+            "idx: {:?} @ {:#x} - new strtab offset: {:#x}",
+            idx, offset, self.sizeof_strtab
+        );
+        // the symbols section reference/index will be the current number of sections
+        let shndx = self.sections.len() + 3; // null + strtab + symtab
+        let section_symbol = SymbolBuilder::new(SymbolType::Section)
+            .section_index(shndx)
+            .create();
+
+        let final_symbol_count =
+            self.special_symbols.len() + self.sections.len() + self.symbols.len() + 1;
+        let section = SectionBuilder::new((final_symbol_count * 4) as u64)
+            .section_type(SectionType::SymTabShndx);
+
+        let mut section = section.name_offset(offset).create(&self.ctx);
+        section.sh_link = 2;
+        // the offset is the head of how many program bits we've added
+        section.sh_offset = self.sizeof_bits as u64;
+        self.sections.insert(
+            idx,
+            SectionInfo {
+                header: section,
+                symbol: section_symbol,
+                name: idx,
+            },
+        );
```

Comment:
I think it would be better if we stored this section separately (i.e. just a `Section`, no section symbol).